fire/SFEquationsMod.F90

```diff
@@ -314,5 +331,139 @@ real(r8) function BackwardRateOfSpread(ros_front, wind_speed)
+ 
+  real(r8) function LengthToBreadth(effective_windspeed, tree_fraction)
+  !
+  !  DESCRIPTION:
+  !  Calculates length to breadth ratio of fire ellipse [unitless], used for calculating area burnt 
+  !
+  !     Canadian Forest Fire Behavior Prediction System Ont.Int.Rep. ST-X-3, 1992
+  !     Information Report GLC-X-10, Wotten et al. 2009
+  !
+  use FatesConstantsMod, only : m_per_km, min_per_hr
+
+  ! ARGUMENTS:
+  real(r8), intent(in) :: effective_windspeed ! effective windspeed [m/min]
+  real(r8), intent(in) :: tree_fraction       ! tree fraction [0-1]
+  
+  ! LOCALS:
+  real(r8) :: windspeed_km_hr ! effective windspeed, converted to correct units [km/hr]
+  
+  ! CONSTANTS:
+  real(r8), parameter :: lb_threshold = 0.55_r8                ! tree canopy fraction below which to use grassland length-to-breadth eqn
+  real(r8), parameter :: m_per_min__to__km_per_hour = 0.06_r8  ! convert wind speed from m/min to km/hr
+
+  windspeed_km_hr = effective_windspeed/m_per_km*min_per_hr
+
+  if (windspeed_km_hr < 1.0_r8) then 
+    LengthToBreadth = 1.0_r8
+ else
+    if (tree_fraction > lb_threshold) then 
+      LengthToBreadth = (1.0_r8 + (8.729_r8* &
+        ((1.0_r8 -(exp(-0.03_r8*m_per_min__to__km_per_hour*effective_windspeed)))**2.155_r8)))
+    else  
+      LengthToBreadth = (1.1_r8*((m_per_min__to__km_per_hour*effective_windspeed)**0.464_r8))
+    endif
+  endif
+
+  end function LengthToBreadth
+
+  !---------------------------------------------------------------------------------------
+  
+  real(r8) function FireSize(length_to_breadth, ros_back, ros_forward, fire_duration)
+    !
+    !  DESCRIPTION:
+    !  Calculates fire size [m2]
+    !
+    !  Eq 14 Arora and Boer JGR 2005 (area of an ellipse)
+    !
+    use FatesConstantsMod, only : pi_const
+
+    ! ARGUMENTS:
+    real(r8), intent(in) :: length_to_breadth ! length to breadth ratio of fire ellipse [unitless]
+    real(r8), intent(in) :: ros_back          ! backwards rate of spread [m/min]
+    real(r8), intent(in) :: ros_forward       ! forward rate of spread [m/min]
+    real(r8), intent(in) :: fire_duration     ! fire duration [min]
+    
+    ! LOCALS:
+    real(r8) :: dist_back    ! distance fire has travelled backwards [m]
+    real(r8) :: dist_forward ! distance fire has travelled forward [m]
+    real(r8) :: fire_size    ! area of fire [m2]
+    
+    dist_back = ros_back*fire_duration
+    dist_forward = ros_forward*fire_duration
+    
+    ! Eq 14 Arora and Boer JGR 2005 (area of an ellipse)
+    if (length_to_breadth < nearzero) then 
+      FireSize = 0.0_r8
+      return
+    end if
+    FireSize = (pi_const/(4.0_r8*length_to_breadth))*((dist_forward + dist_back)**2.0_r8)
+  
+  end function FireSize
+  
+  !---------------------------------------------------------------------------------------
+  
+  real(r8) function AreaBurnt(fire_size, num_ignitions, FDI)
+    !
+    !  DESCRIPTION:
+    !  Calculates area burnt [m2/m2/day]
```

Comment:
Units here don't match what I'd expect based on units in other comments in this function (m2 fire per km2 land).